ablySpec/RealtimeClientConnection.swift

```diff
@@ -1302,6 +1302,44 @@ class RealtimeClientConnection: QuickSpec {
 
             }
 
+            // RTN15
+            context("connection failures once CONNECTED") {
+
+                // RTN15e
+                context("when a connection is resumed") {
+
+                    pending("the connection#key may change and will be provided in the first CONNECTED ProtocolMessage#connectionDetails") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        options.disconnectedRetryTimeout = 1.0
+
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+                        let channel = client.channels.get("test")
+
+                        channel.attach()
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                        client.onDisconnected()
```

Comment:
Why don't you add a `once(.Connecting)` step that explicitly resets the key to another value. That way you can be sure the iOS code is actually updating the `Connection#key`?